src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -1144,6 +1166,120 @@ export default class SourceFileLinter {
 		}
 	}
 
+	#analyzeNewAndApplySettings(node: ts.NewExpression | ts.CallExpression) {
+		node?.arguments?.filter((arg) => ts.isObjectLiteralExpression(arg))
+			.flatMap((arg) => arg.properties)
+			.forEach((prop) => {
+				if (ts.isPropertyAssignment(prop) &&
+					(ts.isCallExpression(node) ||
+						this.#isPropertyBindingType(node, getPropertyNameText(prop.name)))
+				) {
+					if (ts.isObjectLiteralExpression(prop.initializer)) {
+						this.#analyzeModelDataTypes(prop.initializer);
+					} else {
+						this.#analyzeModelDataTypesBinding(prop);
+					}
+				}
+			});
+	}
+
+	#analyzeModelDataTypes(node: ts.ObjectLiteralExpression) {
+		node?.properties.forEach((prop) => {
+			if (!ts.isPropertyAssignment(prop)) {
+				return;
+			}
+
+			// Get the type property
+			let typeField;
+			if (ts.isObjectLiteralExpression(prop.initializer)) {
+				typeField = prop.initializer.properties.find((prop: ts.ObjectLiteralElementLike) => {
+					return ts.isPropertyAssignment(prop) &&
+						ts.isIdentifier(prop.name) &&
+						prop.name.text === "type";
+				});
+			} else if (ts.isStringLiteralLike(prop.initializer) &&
+				ts.isIdentifier(prop.name) && prop.name.text === "type" &&
+				// Whether it's a direct property, named "type" of the Control
+				// or type in property binding
+				!ts.isNewExpression(prop.parent.parent)) {
+				typeField = prop;
+			}
+
+			if (typeField && ts.isPropertyAssignment(typeField)) {
```

Comment:
Can typeField be something else than a PropertyAssignment, in case it has a value?

Reply:
TS resolves it as `ts.ObjectLiteralElementLike | undefined`, so we need this check

Reply:
With the refactoring, now it seem to be working